### PR TITLE
tests: rewrite rebase-submodule .gitmodule file

### LINKED ISSUE #4274: Problem with libgit2's submodules
### Reproduction steps

- Clone the repo.
- Do the `git submodule update --init --recursive` dance.

### Expected behavior

- Everything is fine.

### Actual behavior

```
fatal: No url found for submodule path 'tests/resources/rebase-submodule/my-submodule' in .gitmodules
```

### Version of libgit2 (release number or SHA1)

2270ca9f8feecf93fe39a02a6fbde90a22c1a665

### Operating system(s) tested

macOS 10.12.*, but I don't think it's that relevant 😉.

### PR DESCRIPTION
**Pull request by tiennou**
Fixes #4274

### CODE REVIEW

**Comment by tiennou** [comment]
Crossing fingers (a.k.a thanks Apple, for .DS_Store files) 🔥.

**Comment by pks-t** [comment]
The change looks good to me, thanks. There are more tests with a ".gitmodules" file, though, any particular reason why you didn't adjust them, as well?

```
./tests/resources/submodule_simple/.gitmodules
./tests/resources/submodule_with_path/.gitmodules
./tests/resources/submodules-worktree-parent/.gitmodules
```

**Comment by pks-t** [comment]
Unfortunately, this breaks `rebase::submodule::init_untracked`. I guess we need to create the submodule during the test.

**Comment by tiennou** [comment]
Gah, that gave me a hard time. Tests should be A-OK, but I'm unsure of the cleanliness of my fix. I had to reset after doing the commit, least the index would be weird.

File: tests/rebase/submodule.c [hunk 1]
```diff
@@ -12,9 +13,39 @@ static git_signature *signature;
...
+    cl_git_pass(git_tree_lookup(&tree, repo, &tree_oid));
+
+    git_repository_head(&master_ref, repo);
+    cl_git_pass(git_commit_lookup(&parent, repo, git_reference_target(master_ref)));
+
+    cl_git_pass(git_commit_create_v(&commit_id, repo, git_reference_name(master_ref), signature, signature, NULL, "Fixup .gitmodules", tree, 1, parent));
+
+    /* And a final reset, for good measure */
+    git_object_lookup(&obj, repo, &commit_id, GIT_OBJ_COMMIT);
+    cl_git_pass(git_reset(repo, obj, GIT_RESET_HARD, &opts));
```
Comment (pks-t) [inline tests/rebase/submodule.c]: This misses cleanup of some variables, namely `parent`, `obj` and `master_ref`.

**Comment by tiennou** [comment]
@pks-t Fixed the missing frees, and the space-indent. Would it be possible to have that back-ported to 0.26 as well (this is against master), as it's intended to help downstream users/wrappers ?

**Comment by tiennou** [comment]
Good to go, Travis choked on proxies again.

**Comment by tiennou** [comment]
@pks-t Can this be backported to 0.26.1 too ? Right now Objective-Git tracks "sometimes after 0.26" because of #4274.

**Pull request merged**